pyramid_openapi3/tween.py

```diff
@@ -0,0 +1,53 @@
+"""A tween to validate openapi responses."""
+from .exceptions import ResponseValidationError
+from .wrappers import PyramidOpenAPIRequest
+from .wrappers import PyramidOpenAPIResponse
+from pyramid.httpexceptions import HTTPNotFound
+from pyramid.request import Request
+from pyramid.response import Response
+from pyramid.tweens import reraise
+
+import sys
+import typing as t
+
+
+def response_tween_factory(handler, registry) -> t.Callable[[Request], Response]:
+    """Create response validation tween.
+
+    This tween should run after pyramid exception renderer view, so that
+    final response status and content_type are known and can be validated.
+
+    The only problem here is, that when response validation fails, we have
+    to return some exception response, with an unknown content type.
+    The advantage is, that these are server errors, and if 500 errors are
+    only possible due to response validation errors we don't need to document
+    them in the openapi spec file.
+    """
+
+    def excview_tween(request) -> Response:
+        try:
+            response = handler(request)
+            if not request.environ.get("pyramid_openapi3.validate_response"):
```

Comment:
**pyramid_openapi3.validate_response** infers that only response will not be validated, but the code actually ignores the request too.

Reply:
That's a bit of a source for confusion here... the Request is validated in the view wrapper, but the response is validated after the exception view handler ran (if there is any)